js/api/auth/authFetch.mjs

```
@@ -0,0 +1,33 @@
+import { load } from "../../storage/index.mjs";
+
+export function headers() {
+    const token = load("token");
+
+    return {
+        "Content-Type": "application/json",
+        "Authorization": `Bearer ${token}`
+    };
+}
+
+export async function authFetch(url, options = {}){
+    const requestOptions = {
+        ...options,
+        headers: {
+            ...headers(),
+            ...options.headers
+        }
+    };
+
+    try {
+        const response = await fetch(url, requestOptions);
+        return response;
+    } catch (error) {
+        console.error('Feil under fetch:', error);
+        throw error;
```

`catch block`: to suppress an error
`throw`: to create an error

Understand the differences, they have different purposes. Here you're basically catching an error and throwing it again.

**Suggestion:** remove the throw from the catch block and use the try/catch block in a UI/UX code block e.g event listener.